core/chains/evm/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -251,6 +255,53 @@ AND eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.evm_chain_id = $2`,
 	return errors.Wrap(err, "SetBroadcastBeforeBlockNum failed")
 }
 
+// CheckConfirmedMissingReceipt will attempt to re-send any transaction in the
+// state of "confirmed_missing_receipt". If we get back any type of senderror
+// other than "nonce too low" it means that this transaction isn't actually
+// confirmed and needs to be put back into "unconfirmed" state, so that it can enter
+// the gas bumping cycle. This is necessary in rare cases (e.g. Polygon) where
+// network conditions are extremely hostile.
+//
+// For example, assume the following scenario:
+//
+// 0. We are connected to multiple primary nodes via load balancer
+// 1. We send a transaction, it is confirmed and, we get a receipt
+// 2. A new head comes in from RPC node 1 indicating that this transaction was re-org'd, so we put it back into unconfirmed state
+// 3. We re-send that transaction to a RPC node 2 **which hasn't caught up to this re-org yet**
+// 4. RPC node 2 still has an old view of the chain, so it returns us "nonce too low" indicating "no problem this transaction is already mined"
+// 5. Now the transaction is marked "confirmed_missing_receipt" but the latest chain does not actually include it
+// 6. Now we are reliant on the EthResender to propagate it, and this transaction will not be gas bumped, so in the event of gas spikes it could languish or even be evicted from the mempool and hold up the queue
+// 7. Even if/when RPC node 2 catches up, the transaction is still stuck in state "confirmed_missing_receipt"
+//
+// This scenario might sound unlikely but has been observed to happen multiple times in the wild on Polygon.
+func (ec *EthConfirmer) CheckConfirmedMissingReceipt(ctx context.Context) (err error) {
+	var attempts []EthTxAttempt
+	err = ec.q.Select(&attempts,
+		`SELECT DISTINCT ON (eth_tx_id) eth_tx_attempts.*
+		FROM eth_tx_attempts
+		JOIN eth_txes ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.state = 'confirmed_missing_receipt'
+		WHERE evm_chain_id = $1
+		ORDER BY eth_tx_attempts.eth_tx_id ASC, eth_tx_attempts.gas_price DESC, eth_tx_attempts.gas_tip_cap DESC`,
+		ec.chainID.String())
+	if err != nil {
+		return err
+	}
+	reqs, _ := batchSendTransactions(attempts, int(ec.config.EvmRPCDefaultBatchSize()), ec.lggr, ec.ctx, ec.ethClient)
+	var ethTxIDsToUnconfirm []int64
+	for idx, req := range reqs {
+		// Add to Unconfirm array, all tx where error wasn't NonceTooLow.
+		if req.Error == nil || !evmclient.NewSendError(req.Error).IsNonceTooLowError() {
+			ethTxIDsToUnconfirm = append(ethTxIDsToUnconfirm, attempts[idx].EthTxID)
+		}
+	}
+	_, err = ec.q.Exec(`UPDATE eth_txes SET state='unconfirmed' WHERE id = ANY($1)`, pq.Array(ethTxIDsToUnconfirm))
```

Comment:
We should find the highest one and update that nonce and everything below it.

e.g. if nonce 43 is unconfirmed then nonce 42, 41 etc are definitely also unconfirmed.

Reply:
If tx with nonce 43 is unconfirmed then there is absolutely no possible way a tx with nonce lower than that can possibly be confirmed. So we might as well mark the all unconfirmed. Make sense?

Reply:
Sorry I had it backwards. Meant to say e.g. if nonce 41 is unconfirmed then its not possible for 42, 43 etc to be confirmed

Reply:
On second thoughts, lets keep it how you have it here.

Reply:
Had a few offline discussions about this. My views on this are:

`I think trying to mark higher nonces as unconfirmed based on a lower nonce being unconfirmed is much complex, and giving almost no benefit. The higher nonces would already be present in the same batch here, if it was earlier marked as confirmed_missing_receipt anyways. So they would already be marked correctly in this same query.`

What do you think?